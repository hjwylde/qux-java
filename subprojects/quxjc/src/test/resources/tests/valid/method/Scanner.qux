package valid.method

type Scanner is {str input, int index}

str Scanner::next():
    return this.input[this.index++]

str Scanner::peek():
    return this.input[this.index]

void Scanner::reset():
    this.index = 0

void main([str] args):
    scanner = {input: 'foo bar baz', index: 0}
    print scanner
    print scanner::next()
    print scanner::peek()
    print scanner
    scanner::next()
    print scanner
    scanner::reset()
    print scanner::next()

