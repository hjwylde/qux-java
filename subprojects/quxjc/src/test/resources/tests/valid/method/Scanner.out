{index: 0, input: foo bar baz}
f
o
{index: 1, input: foo bar baz}
{index: 2, input: foo bar baz}
f
